packages/strapi-generate-new/lib/merge-template.js

```diff
@@ -0,0 +1,164 @@
+}
+
+function getRepoInfo(githubURL) {
+  // Make sure it's a github url
+  const address = githubURL.split('://')[1];
+  if (!address.startsWith('github.com')) {
+    throw Error('A Strapi template can only be a GitHub URL');
+  }
+
+  // Parse github address into parts
+  const [, username, name, ...rest] = address.split('/');
+  const isRepo = username != null && name != null;
+  if (!isRepo || rest.length > 0) {
+    throw Error('A template URL must be the root of a GitHub repository');
+  }
+
+  return { username, name };
+}
+
+async function downloadGithubRepo(repoInfo, rootPath) {
+  const { username, name, branch = 'master' } = repoInfo;
+  const codeload = `https://codeload.github.com/${username}/${name}/zip/${branch}`;
+  const templatePath = path.resolve(rootPath, 'tmp-template');
+  const response = await fetch(codeload);
+  await new Promise(resolve => {
+    response.body.pipe(unzip.Extract({ path: templatePath })).on('close', resolve);
+  });
+
+  return templatePath;
+}
+
+// Merge the template's template.json into the Strapi project's package.json
+async function mergePackageJSON(rootPath, templatePath, repoInfo) {
+  // Import the package.json and template.json templates
+  const packageJSON = require(path.resolve(rootPath, 'package.json'));
+  const templateJSON = require(path.resolve(templatePath, 'template.json'));
+
+  // Make sure the template.json doesn't overwrite the UUID
+  if (templateJSON.strapi && templateJSON.strapi.uuid) {
+    throw Error('A template cannot overwrite the Strapi UUID');
+  }
+
+  // Use lodash to deeply merge them
+  const mergedConfig = _.merge(packageJSON, templateJSON);
+
+  // Prefix Strapi UUID with starter info
+  const prefix = `STARTER:${repoInfo.username}/${repoInfo.name}:`;
+  mergedConfig.strapi = {
+    uuid: prefix + mergedConfig.strapi.uuid,
+  };
+
+  // Save the merged config as the new package.json
+  await fse.writeJSON(path.resolve(rootPath, 'package.json'), mergedConfig, {
+    spaces: 2,
+  });
+}
+
+// Merge all allowed files and directories
+async function mergeFilesAndDirectories(rootPath, templatePath) {
+  const copyPromises = fse
+    .readdirSync(path.resolve(templatePath, 'template'))
+    .map(async fileOrDir => {
+      // Copy the template contents from the downloaded template to the Strapi project
+      await fse.copy(
+        path.resolve(templatePath, 'template', fileOrDir),
+        path.resolve(rootPath, fileOrDir),
+        { overwrite: true, recursive: true }
+      );
+    });
+  await Promise.all(copyPromises);
+}
+
+module.exports = async function mergeTemplate(templateUrl, rootPath) {
+  // Download template repository to a temporary directory
+  const repoInfo = getRepoInfo(templateUrl);
+  console.log(`Installing ${repoInfo.username}/${repoInfo.name} template.`);
+  const templateParentPath = await downloadGithubRepo(repoInfo, rootPath);
+  const templatePath = path.resolve(templateParentPath, fse.readdirSync(templateParentPath)[0]);
```

Comment:
You should defined a templatePath and downloadTheRepo inside of it not return the path from the download function. And Then use it. The fs.readdeirSync()[0] should not be necessary